sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/filters/TimeWindowFilter.java

```diff
@@ -30,38 +34,83 @@ public final class TimeWindowFilter implements FeatureFilter {
      */
     @Override
     public boolean evaluate(FeatureFilterEvaluationContext context) {
-        String start = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_START);
-        String end = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_END);
+        final TimeWindowFilterParameters filterParameters = new TimeWindowFilterParameters(context);
+        if (!filterParameters.isValid()) {
+            return false;
+        }
 
         ZonedDateTime now = ZonedDateTime.now();
-
-        if (!StringUtils.hasText(start) && !StringUtils.hasText(end)) {
-            LOGGER.warn("The {} feature filter is not valid for feature {}. It must specify either {}, {}, or both.",
-                this.getClass().getSimpleName(), context.getName(), TIME_WINDOW_FILTER_SETTING_START,
-                TIME_WINDOW_FILTER_SETTING_END);
+        // check if match start time(end time) when specify start time(end time)
+        if (filterParameters.startTime != null && now.isBefore(filterParameters.startTime) ||
+            (filterParameters.endTime != null && now.isAfter(filterParameters.endTime))) {
             return false;
         }
-        
-        ZonedDateTime startTime = null;
-        ZonedDateTime endTime = null;
 
-        try {
-            startTime = StringUtils.hasText(start)
-                ? ZonedDateTime.parse(start, DateTimeFormatter.ISO_DATE_TIME)
-                : null;
-            endTime = StringUtils.hasText(end)
-                ? ZonedDateTime.parse(end, DateTimeFormatter.ISO_DATE_TIME)
-                : null;
-        } catch (DateTimeParseException e) {
-            startTime = StringUtils.hasText(start)
-                ? ZonedDateTime.parse(start, DateTimeFormatter.RFC_1123_DATE_TIME)
-                : null;
-            endTime = StringUtils.hasText(end)
-                ? ZonedDateTime.parse(end, DateTimeFormatter.RFC_1123_DATE_TIME)
-                : null;
+        // Need to set Zone Offset when compare with crontab expression
+        boolean enabled = filterParameters.filterCronTabExpressions.size() == 0;
```

Comment:
We don't need to check the rest of this if this is true, we can just return true here.

Reply:
Yes. Return true when no filters